core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -70,7 +70,10 @@ private[deploy] class Worker(
   if (conf.get(config.DECOMMISSION_ENABLED)) {
     logInfo("Registering SIGPWR handler to trigger decommissioning.")
     SignalUtils.register("PWR", "Failed to register SIGPWR handler - " +
-      "disabling worker decommission feature.")(decommissionSelf)
+      "disabling worker decommission feature.") {
+       self.send(WorkerSigPWRReceived)
```

Comment:
I think this might mean we return from handling the signal right away rather than waiting for decommissionSelf to be finished. Is this an intentional change?

Also this will no longer report a decommissioning failure by signal return value, so may block pod deletion or other cleanup task longer than needed.

Reply:
This's Worker, I guess you cares more about the executor? In Worker, `decommissionSelf` always returns true. and In exeutor, there's a change to return false to fail the decommissionSelf but seems rarely happen. If you would insist on returning the value, I think we can use askSync instead.

Reply:
Can you look into what the difference of this behavior might cause at the system level and then tell me if that’s a desired change? I’m ok with us making changes here, I just want us to be intentional and know if we need to test the change and it seems like this change was incidental.

Reply:
The return value of the signal handling decides whether we should forward the signal to the other handlers. If `true`, no other handlers will handle the `PWR` signal except ourselves. If `false`, we will handle it (for decommission) and other handlers will handle it too.  Do you expect other handlers to continue handling the `SIGPWR` when the system isn't really experiencing a power failure?

Reply:
I do, I think if the signal is unhandled then the process will be killed immediately. If we think of decommissioning/graceful shut down I believe that behavior is desirable, since if we can't shut down gracefully the least we can do is exit quickly.

Reply:
Updated, but please note that I only updated for the executor's case since Worker's case always returns `true` before.